src/utils/test-page-url.js

```diff
@@ -0,0 +1,46 @@
+import path from 'path';
+import OS from 'os-family';
+
+const PROTOCOL_RE           = /^([\w-]+?)(?=\:)/;
+const SUPPORTED_PROTOCOL_RE = /^(https?|file):/;
+const IMPLICIT_PROTOCOL_RE  = /^\/\//;
+const ABSOLUTE_PATH_RE      = /^\/[^\/]/;
+const WIN_ABSOLUTE_PATH_RE  = /^\w:(\/|\\)/;
+const RELATIVE_PATH_RE      = /^\.\.?(\/|\\)/;
+
+
+function isAbsolutePath (url) {
+    return OS.win ? WIN_ABSOLUTE_PATH_RE.test(url) : ABSOLUTE_PATH_RE.test(url);
+}
+
+function resolveFileUrl (url, testFileName) {
+    var testFileDir = path.dirname(testFileName);
+
+    if (RELATIVE_PATH_RE.test(url))
+        url = path.join(testFileDir, url);
+
+    return 'file://' + url;
+}
+
+export function validateUrl (url) {
+    var protocol = url.match(PROTOCOL_RE);
+
+    if (protocol && !SUPPORTED_PROTOCOL_RE.test(url) &&
+        !(OS.win && WIN_ABSOLUTE_PATH_RE.test(url)) &&
```

Comment:
No need for multi line conditions. Just refactor it into variables with meaningful names